### DISCUSSION TITLE
What is the use of the `commons` chunk given that we have the dynamically generated `shared` chunks?

### DISCUSSION THREAD

**Discussion by b4stien** [gatsbyjs/gatsby#31078] [Help]
In its Webpack config Gatsby specifies a `commons` chunk where it puts every module that is used by every pages/templates. The specific config is: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/webpack.config.js#L598-L603.

In addition there is also `shared` chunks which dynamically regroups every module used by at least two pages/templates. The config is: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/utils/webpack.config.js#L605-L620.

Hence the question: why do we need both? Isn't the `commons` chunk a special case of a `shared` chunk which concerns every pages/templates? Is there something I don't understand and which explains the special case?

**Comment by LekoArts** [comment]
`commons` is when it's used in **all** components, `shared` is when it's used in at least 2 components **but** not in all components. So yeah, you could say it's a special case. The PR adding it (with explanation) is here: https://github.com/gatsbyjs/gatsby/pull/22253
